### DISCUSSION TITLE
Ejabberd 22.10 mod_conversejs

### DISCUSSION THREAD

**Discussion by AndreyKaO** [processone/ejabberd#3951] [Questions] [answered]
Good afternoon. We installed ejabberd 22.10, connected the converse_js module, but we cannot change the language of the main modules, that is, groupchats, contacts, im a online. There were no such problems on version 22.05, the language was automatically substituted for the current one. Requires for example Russian
![Снимок](https://user-images.githubusercontent.com/98833421/203318431-8757f974-0b91-4ccc-b3d6-aa81e1f7d2ba.JPG)

**Comment by prefiks** [comment]
I just added e2779e11552d2e2b7dcf8a28576a1f1bee82fdb9 that should fix this this when using converse files from cdn (you don't have `convesejs_resource` option), if you have that option then i guess your local copy is missing locale files (like `/locales/ru-LC_MESSAGES-converse-po.js`).

**Comment by badlop** [comment, marked as answer]
Are you sure 22.05 worked correctly?

I think the problem is that mod_conversejs.erl sets the option [i18n for Conversejs](https://conversejs.org/docs/html/configuration.html#i18n). And when that is set, the browser language is not considered... because i18n option is enforced! That was introduced in f461bcb5973bd732b41b82a60ed07b9c8a7ef208

I've just committed a fix in daaaf221cbeedcd5156fb098efcc9eacd82550e0, simply removing the line that set i18n option.

In your case, you have simple solutions that don't require compiling ejabberd:

A) Set i18n option to "ru", but remember this language will be enforced to all users:
```yaml
modules:
  mod_conversejs:
    conversejs_options:
      i18n: ru
```

B) Or set i18n option to some dummy and non-existing language, that way Converse will use the browser language of the user:
```yaml
modules:
  mod_conversejs:
    conversejs_options:
      i18n: dummy
```
